streams/src/main/java/org/apache/kafka/streams/kstream/internals/RepartitionNode.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+
+class RepartitionNode<K, V> extends StatelessProcessorNode<K, V> {
+
+    private final Serde<K> keySerde;
+    private final Serde<V> valueSerde;
+    private final String sinkName;
+    private final String sourceName;
+    private final String repartitionTopic;
+    private final String processorName;
+
+
+    RepartitionNode(final String predecessorNodeName,
+                    final String nodeName,
+                    final String sourceName,
+                    final ProcessorSupplier<K, V> processorSupplier,
+                    final Serde<K> keySerde,
+                    final Serde<V> valueSerde,
+                    final String sinkName,
+                    final String repartitionTopic,
+                    final String processorName) {
+
+        super(predecessorNodeName,
+              nodeName,
+              processorSupplier,
+              false);
+
+        this.keySerde = keySerde;
+        this.valueSerde = valueSerde;
+        this.sinkName = sinkName;
+        this.sourceName = sourceName;
+        this.repartitionTopic = repartitionTopic;
+        this.processorName = processorName;
+    }
+
+    Serde<K> keySerde() {
+        return keySerde;
+    }
+
+    Serde<V> valueSerde() {
+        return valueSerde;
+    }
+
+    String sinkName() {
+        return sinkName;
+    }
+
+    String sourceName() {
+        return sourceName;
+    }
+
+    String repartitionTopic() {
+        return repartitionTopic;
+    }
+
+    String processorName() {
+        return processorName;
+    }
+
+    @Override
+    void writeToTopology(InternalTopologyBuilder topologyBuilder) {
+        //TODO will implement in follow-up pr
+    }
+
+    static <K, V> RepartitionNodeBuilder<K, V> repartitionNodeBuilder() {
+        return new RepartitionNodeBuilder<>();
+    }
+
+
+    static final class RepartitionNodeBuilder<K, V> {
```

Comment:
As I mentioned in the other place, if the `XXBuilder.build()` call will only be triggered once in its life time, I'd suggest just keeping the `RepartitionNode` with non-final parameters for simplicity.